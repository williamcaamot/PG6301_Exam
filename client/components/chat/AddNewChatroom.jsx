import React, { useState } from "react";
import ErrorMessage from "../globals/ErrorMessage.jsx";
import SuccessMessage from "../globals/SuccessMessage.jsx";
import { METHODS } from "stream-http";
import { Link } from "react-router-dom";

function AddNewChatroom() {
  const [chatRoomTitle, setChatRoomTitle] = useState("");
  const [chatRoomDescription, setChatRoomDescription] = useState("");

  const [errorMessage, setErrorMessage] = useState();
  const [successMessage, setSuccessMessage] = useState();

  async function handleAddChatRoom(e) {
    e.preventDefault();
    try {
      const chatroom = {
        title: chatRoomTitle,
        description: chatRoomDescription,
      };
      const res = await fetch("/api/v1/chatroom", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(chatroom),
      });
      const { data, message } = await res.json();
      if (res.status !== 201) {
        //TODO Make this logic better
        setSuccessMessage(null);
        setErrorMessage(message || "En ukjent feil har oppst√•tt");
        return;
      }
      setErrorMessage(null);
      setSuccessMessage(message);
    } catch (e) {
      setSuccessMessage(null);
      setErrorMessage(e.message);
    }
  }

  return (
    <>
      <div className={"pageContentWrapper"}>
        <div className={"innerWrapper"}>
          <h2>Add new chat room</h2>
          <div style={{ width: "100%", display: "flex" }}>
            <form onSubmit={handleAddChatRoom}>
              <p>
                Name of chatroom:{" "}
                <input
                  placeholder={"Title"}
                  value={chatRoomTitle}
                  onChange={(e) => setChatRoomTitle(e.target.value)}
                />
              </p>
              <br />
              <p>
                Description of chatroom:{" "}
                <input
                  placeholder={"What is the chatroom about?"}
                  value={chatRoomDescription}
                  onChange={(e) => setChatRoomDescription(e.target.value)}
                />
              </p>
              <br />
              <button>Legg til chatroom</button>
            </form>
          </div>
          <ErrorMessage message={errorMessage} />
          <SuccessMessage message={successMessage} />
        </div>
      </div>
    </>
  );
}
export default AddNewChatroom;
