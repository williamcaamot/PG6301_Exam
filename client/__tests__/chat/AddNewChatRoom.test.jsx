import renderer, { act } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import AddNewChatroom from "../../components/chat/AddNewChatroom.jsx";

describe("Add new chat room", () => {
  it("Should render page for adding new chat rooms", () => {
    async function setChatroom() {}

    const component = renderer.create(
      <MemoryRouter>
        <AddNewChatroom />
      </MemoryRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
