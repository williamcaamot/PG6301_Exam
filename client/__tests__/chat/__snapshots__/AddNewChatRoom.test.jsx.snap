// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add new chat room Should render page for adding new chat rooms 1`] = `
<div
  className="pageContentWrapper"
>
  <div
    className="innerWrapper"
  >
    <h2>
      Add new chat room
    </h2>
    <div
      style={
        {
          "display": "flex",
          "width": "100%",
        }
      }
    >
      <form
        onSubmit={[Function]}
      >
        <p>
          Name of chatroom:
           
          <input
            onChange={[Function]}
            placeholder="Title"
            value=""
          />
        </p>
        <br />
        <p>
          Description of chatroom:
           
          <input
            onChange={[Function]}
            placeholder="What is the chatroom about?"
            value=""
          />
        </p>
        <br />
        <button
          onClick={[Function]}
        >
          Legg til chatroom
        </button>
      </form>
    </div>
  </div>
</div>
`;
